### PR TITLE
[Merged by Bors] - feat: More `sup_indep` lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
Match https://github.com/leanprover-community/mathlib/pull/11932

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by riccardobrasca** [comment]
LGTM, but there is a linter error.

bors d+

File: Mathlib/Order/SupIndep.lean [hunk 1]
```diff
@@ -4,12 +4,13 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
 
 ! This file was ported from Lean 3 source module order.sup_indep
-! leanprover-community/mathlib commit 1c857a1f6798cb054be942199463c2cf904cb937
+! leanprover-community/mathlib commit c4c2ed622f43768eff32608d4a0f8a6cec1c047d
 ! Please do not edit these lines, except to modify the commit id
 ! if you have ported upstream changes.
 -/
 import Mathlib.Data.Finset.Pairwise
 import Mathlib.Data.Finset.Powerset
+import Mathlib.Data.Finset.Sigma
```
Comment (eric-wieser) [inline Mathlib/Order/SupIndep.lean:13]: This is not the same import order as mathlib3

**Review by eric-wieser — APPROVED** [review]
One minor comment about imports; note that changing the import order can in principle affect behavior.

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**